pkgs/development/libraries/gtk/patches/3.0-sysconfdir_install.patch

```diff
@@ -0,0 +1,45 @@
+diff --git a/meson.build b/meson.build
+index aed48fc3f6..33632d3ecf 100644
+--- a/meson.build
++++ b/meson.build
+@@ -176,6 +176,11 @@ gtk_libdir = join_paths(gtk_prefix, get_option('libdir'))
+ gtk_datadir = join_paths(gtk_prefix, get_option('datadir'))
+ gtk_localedir = join_paths(gtk_prefix, get_option('localedir'))
+ gtk_sysconfdir = join_paths(gtk_prefix, get_option('sysconfdir'))
++if get_option('sysconfdir_install') != ''
++  gtk_sysconfdir_install = join_paths(gtk_prefix, get_option('sysconfdir_install'))
++else
++  gtk_sysconfdir_install = sysconfdir
++endif
+ gtk_applicationsdir = join_paths(gtk_datadir, 'applications')
+ gtk_schemasdir = join_paths(gtk_datadir, 'glib-2.0/schemas')
+ gtk_appdatadir = join_paths(gtk_datadir, 'metainfo')
+@@ -751,7 +756,7 @@ immodules = [
+   [ 'ipa', files([ immodule_srcdir + 'imipa.c' ]) ],
+   [ 'multipress', files( [ immodule_srcdir + 'gtkimcontextmultipress.c', immodule_srcdir + 'immultipress.c' ]), true,
+                     ['-DMULTIPRESS_LOCALEDIR="@0@"'.format(gtk_mplocaledir),
+-                     '-DMULTIPRESS_CONFDIR="@0@/gtk-3.0"'.format(gtk_sysconfdir)] ],
++                     '-DMULTIPRESS_CONFDIR="@0@/gtk-3.0"'.format(gtk_sysconfdir_install)] ],
```

Comment:
Does GTK handle the absence of the config file gracefully? Otherwise the input method will be broken without `environment.etc."gtk-3.0/im-multipress.conf".source = "${pkgs.gtk3}/etc/gtk-3.0/im-multipress.conf";`. (Though I am not sure if the input method used by anyone.)

Reply:
The path of this file gets hardcoded in the shared object and gtk tries to install the config to that path and read from it. For this single file nothing changed and you still can't easily change it.

Reply:
But the file will not exist in `/etc` unless user explicitly creates it, which most people won’t, thus breaking the input method for them. And I am not really convinced it makes sense to to edit the input method description in the first place.

Reply:
It is not installing into ``/etc`` but ``$out/etc``.

```
 ➜ strings result/lib/gtk-3.0/3.0.0/immodules/im-multipress.so | rg im-multipress.conf
/nix/store/nbigklvq9nimb9azglq76zyhkwlyhdik-gtk+3-3.24.35/etc/gtk-3.0/im-multipress.conf
```

Reply:
Sorry, you are right, I misread it.